src/receive_imf.rs

```diff
@@ -167,7 +166,6 @@ pub(crate) async fn receive_imf_inner(
         } else {
             Origin::IncomingUnknownTo
         },
-        prevent_rename,
```

Comment:
I don't think there is a point in preventing renaming of displaynames in the To: header, at least I can't remember any mailing list messages that had weird displaynames in the To header; back when I introduced this `prevent_rename`, this was mainly about `From` headers (like, `From: adb <notifications@github.com>`).